lib/crimp.rb

```diff
@@ -1,74 +1,43 @@
-require 'crimp/version'
-require 'digest'
-
-class Numeric
-  # see http://patshaughnessy.net/2014/1/9/how-big-is-a-bignum
-  def bignum?
-    self >= 4611686018427387904
-  end
-end
-
-module Crimp
-  def self.signature(obj)
-    Digest::MD5.hexdigest stringify(obj)
-  end
-
-  def self.stringify(obj)
-    convert(obj).tap { |o| return o.class == String ? o : to_string(o) }
-  end
+# frozen_string_literal: true
 
-  private
-
-  def self.convert(obj)
-    case obj
-    when Array
-      parse_array obj
-    when Hash
-      parse_hash obj
-    when String
-      obj
-    else
-      to_string obj
+require 'digest'
+class Crimp
+  class << self
+    def signature(obj)
+      Digest::MD5.hexdigest(to_s(obj))
     end
-  end
 
-  def self.hash_to_array(hash)
-    [].tap do |a|
-      hash.each { |k, v| a << pair_to_string(k, v) }
+    def to_s(obj)
+      to_a(obj).flatten.join
     end
-  end
 
-  def self.pair_to_string(k, v)
-    "#{stringify k}=>#{stringify v}"
-  end
-
-  def self.parse_array(array)
-    array.map { |e| stringify(e) }.sort
-  end
+    def to_a(obj)
+      obj = coerce(obj)
+
+      case obj
+      when String
+        [obj, 's']
+      when Numeric
+        [obj, 'n']
+      when TrueClass, FalseClass
+        [obj, 'b']
+      when NilClass
+        [nil, '_']
+      when Enumerable
+        process_collection(obj)
+      else
+        raise ArgumentError
```

Comment:
Should we say we don't support the type being parsed?

Reply:
I'm already on it